commands/user_test.go

```diff
@@ -52,3 +52,284 @@ func (s *MmctlUnitTestSuite) TestSearchUserCmd() {
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetUserByUsername(arg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetUser(arg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		err := userDeactivateCmdF(s.client, &cobra.Command{}, []string{arg})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal("Unable to find user '"+arg+"'", printer.GetErrorLines()[0])
+	})
+
+	s.Run("Delete multiple users", func() {
+		printer.Clean()
+		mockUser1 := model.User{Id: "userId1", Email: "user1@example.com", Username: "user1"}
+		mockUser2 := model.User{Id: "userId2", Email: "user2@example.com", Username: "user2"}
+		mockUser3 := model.User{Id: "userId3", Email: "user3@example.com", Username: "user3"}
+
+		argEmails := []string{mockUser1.Email, mockUser2.Email, mockUser3.Email}
+		argUsers := []model.User{mockUser1, mockUser2, mockUser3}
+
+		for i := 0; i < len(argEmails); i++ {
+			s.client.
+				EXPECT().
+				GetUserByEmail(argEmails[i], "").
+				Return(&argUsers[i], &model.Response{Error: nil}).
+				Times(1)
+		}
+
+		for i := 0; i < len(argEmails); i++ {
+			s.client.
+				EXPECT().
+				DeleteUser(argUsers[i].Id).
+				Return(true, &model.Response{Error: nil}).
+				Times(1)
+		}
+
+		err := userDeactivateCmdF(s.client, &cobra.Command{}, argEmails)
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Delete multiple users with argument mixture of emails usernames and userIds", func() {
+		printer.Clean()
+		mockUser1 := model.User{Id: "userId1", Email: "user1@example.com", Username: "user1"}
+		mockUser2 := model.User{Id: "userId2", Email: "user2@example.com", Username: "user2"}
+		mockUser3 := model.User{Id: "userId3", Email: "user3@example.com", Username: "user3"}
+
+		argsDelete := []string{mockUser1.Id, mockUser2.Email, mockUser3.Username}
+		argUsers := []model.User{mockUser1, mockUser2, mockUser3}
+
+		// mockUser1
+		s.client.
+			EXPECT().
+			GetUserByEmail(argsDelete[0], "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetUserByUsername(argsDelete[0], "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetUser(argsDelete[0], "").
+			Return(&argUsers[0], &model.Response{Error: nil}).
+			Times(1)
+
+		// mockUser2
+		s.client.
+			EXPECT().
+			GetUserByEmail(argsDelete[1], "").
+			Return(&argUsers[1], &model.Response{Error: nil}).
+			Times(1)
+
+		// mockUser3
+		s.client.
+			EXPECT().
+			GetUserByEmail(argsDelete[2], "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetUserByUsername(argsDelete[2], "").
+			Return(&argUsers[2], &model.Response{Error: nil}).
+			Times(1)
+
+		for i := 0; i < len(argsDelete); i++ {
```

Comment:
nit: you can iterate over `argsUsers` to improve readability:

```suggestion
		for _, user := range argUsers {
```